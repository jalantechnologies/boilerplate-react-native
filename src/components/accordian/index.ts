export { default as Accordion } from './accordian';
export { default as AccordionItem } from './accordianItem';
export type { AccordionProps, AccordionItemProps } from './types';
