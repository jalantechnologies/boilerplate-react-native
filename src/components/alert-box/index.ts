export { AlertProvider, useAlert } from './AlertProvider';
export { AlertType } from '../../types/alert';
