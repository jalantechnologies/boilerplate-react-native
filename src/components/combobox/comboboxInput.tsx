import React from 'react';
import { TextInput, TouchableOpacity, View } from 'react-native';

import { comboboxStyles } from './styles';

type ComboboxInputProps = {
  value: string;
  placeholder?: string;
  isOpen: boolean;
  onFocus: () => void;
  onChangeText: (text: string) => void;
};

export const ComboboxInput = ({
  value,
  placeholder = 'Select...',
  isOpen,
  onFocus,
  onChangeText,
}: ComboboxInputProps) => (
  <TouchableOpacity onPress={onFocus} activeOpacity={0.8}>
    <View style={comboboxStyles.inputContainer}>
      <TextInput
        style={comboboxStyles.input}
        value={value}
        placeholder={placeholder}
        onFocus={onFocus}
        onChangeText={onChangeText}
        editable={!isOpen}
      />
    </View>
  </TouchableOpacity>
);

ComboboxInput.defaultProps = {
  placeholder: 'Select...',
};
