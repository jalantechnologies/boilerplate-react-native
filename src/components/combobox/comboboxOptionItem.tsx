import React from 'react';
import { TouchableOpacity, Text } from 'react-native';

import { comboboxStyles } from './styles';
import { ComboboxOption } from './types';

export const ComboboxOptionItem = ({
  item,
  onPress,
}: {
  item: ComboboxOption;
  onPress: (option: ComboboxOption) => void;
}) => (
  <TouchableOpacity style={comboboxStyles.option} onPress={() => onPress(item)}>
    <Text style={comboboxStyles.optionText}>{item.label}</Text>
  </TouchableOpacity>
);
