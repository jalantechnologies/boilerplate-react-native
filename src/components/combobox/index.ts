export { ComboboxInput } from './comboboxInput';
export { ComboboxOptionList } from './comboboxOptionList';
export { ComboboxOptionItem } from './comboboxOptionItem';
export * from './hooks/useCombobox';
export * from './types';
export * from './styles';
