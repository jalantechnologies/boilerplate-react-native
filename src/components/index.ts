export { default as Brand } from './brand/brand';
export { default as ErrorFallback } from './error-fallback/error-fallback';
export { default as OTPInput } from './otp-input/otp-input';
