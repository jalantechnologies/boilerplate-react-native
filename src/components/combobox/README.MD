# Combobox Component

A customizable autocomplete input with async support and multi-select capabilities.

## Features
- Virtualized list for performance
- Async options loading
- Multi-select support
- Debounced search
- Type-safe implementation

## Props

| Prop | Type | Default | Description |
|------|------|---------|-------------|
| `options` | `ComboboxOption[]` | `[]` | Static options list |
| `value` | `ComboboxOption[]` | `[]` | Controlled selected values |
| `onChange` | `(options: ComboboxOption[]) => void` | - | Selection change handler |
| `async` | `boolean` | `false` | Enable async mode |
| `multiselect` | `boolean` | `false` | Allow multiple selections |
| `debounceMs` | `number` | `300` | Debounce timeout for async search |
| `loadOptions` | `(query: string) => Promise<ComboboxOption[]>` | - | Async options loader |

## Basic Usage

```tsx
import { Combobox } from './components/Combobox';

const App = () => (
  <Combobox
    options={[
      { value: '1', label: 'Option 1' },
      { value: '2', label: 'Option 2' }
    ]}
    onChange={(selected) => console.log(selected)}
    placeholder="Choose option"
  />
);

## Async search

```tsx
<Combobox
  async
  loadOptions={async (query) => {
    const response = await fetch(`/api/search?q=${query}`);
    return response.json();
  }}
  debounceMs={500}
/>
```
## Multi select

```tsx
 <Combobox
  multiselect
  options={mockOptions}
  onChange={(selected) => updateState(selected)}
/>
```
