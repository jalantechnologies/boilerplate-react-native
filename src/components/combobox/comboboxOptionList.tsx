import React from 'react';
import { FlatList } from 'react-native';

import { ComboboxOptionItem } from './comboboxOptionItem';
import { comboboxStyles } from './styles';
import { ComboboxOption } from './types';

const ITEM_HEIGHT = 40;

export const ComboboxOptionList = ({
  options,
  handleSelect,
}: {
  options: ComboboxOption[];
  handleSelect: (option: ComboboxOption) => void;
}) => (
  <FlatList
    data={options}
    style={comboboxStyles.list}
    keyExtractor={item => item.value}
    getItemLayout={(_, index) => ({
      length: ITEM_HEIGHT,
      offset: ITEM_HEIGHT * index,
      index,
    })}
    renderItem={({ item }) => <ComboboxOptionItem item={item} onPress={handleSelect} />}
    keyboardShouldPersistTaps="handled"
  />
);
