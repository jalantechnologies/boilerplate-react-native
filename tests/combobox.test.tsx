import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';

import { ComboboxInput } from '../src/components/combobox'; // Correct import path

describe('ComboboxInput Component', () => {
  const mockProps = {
    value: '',
    placeholder: 'Search...',
    isOpen: false,
    onFocus: jest.fn(),
    onChangeText: jest.fn(),
  };

  it('renders correctly with placeholder', () => {
    const { getByPlaceholderText } = render(<ComboboxInput {...mockProps} />);
    expect(getByPlaceholderText('Search...')).toBeTruthy();
  });

  it('triggers onFocus when pressed', () => {
    const { getByPlaceholderText } = render(<ComboboxInput {...mockProps} />);
    fireEvent.press(getByPlaceholderText('Search...'));
    expect(mockProps.onFocus).toHaveBeenCalled();
  });

  it('updates input value', () => {
    const { getByDisplayValue } = render(<ComboboxInput {...mockProps} value="test" />);
    expect(getByDisplayValue('test')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const tree = render(<ComboboxInput {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
