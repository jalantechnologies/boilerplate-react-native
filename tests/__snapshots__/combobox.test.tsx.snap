// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboboxInput Component matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "borderColor": "#e5e7eb",
        "borderRadius": 8,
        "borderWidth": 1,
        "padding": 12,
      }
    }
  >
    <TextInput
      editable={true}
      onChangeText={[MockFunction]}
      onFocus={
        [MockFunction] {
          "calls": [
            [],
          ],
          "results": [
            {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      placeholder="Search..."
      style={
        {
          "fontSize": 16,
        }
      }
      value=""
    />
  </View>
</View>
`;
